data/covering_grammar/lib/error_analysis.py

```diff
@@ -19,51 +19,58 @@
+import prettytable
 import pynini
+from pynini.lib import rewrite
 
 
 def main(args: argparse.Namespace) -> None:
-    cg_fst = pynini.string_file(args.cg_path).closure().optimize()
-    rulematch_predmatch = 0
-    rulematch_pred_notmatch = 0
-    not_rulematch_predmatch = 0
-    not_rulematch_pred_notmatch = 0
-    total_records = 0
-    with open(args.test_path, "r") as source:
-        for line in source:
-            total_records += 1
-            parts = line.split("\t")
-            lg = parts[0].rstrip()
-            act = parts[1].replace(" ", "").replace(".", "").rstrip()
-            pred_pron = parts[2].replace(" ", "").replace(".", "").rstrip()
-            # TODO: use Pynini's rewrite module here.
-            lattice = lg @ cg_fst @ pred_pron
-            lattice.project("output")
-            if lattice.start() == pynini.NO_STATE_ID:
-                if act == pred_pron:
-                    not_rulematch_predmatch += 1
+    with pynini.default_token_type("utf8"):
+        cg_fst = pynini.string_file(args.cg_path).closure().optimize()
+        rulematch_predmatch = 0
+        rulematch_pred_notmatch = 0
+        not_rulematch_predmatch = 0
+        not_rulematch_pred_notmatch = 0
+        total_records = 0
+        with open(args.test_path, "r") as source:
+            for line in source:
+                total_records += 1
+                (ortho_str, hypo_p_str, gold_p_str) = line.rstrip().split(
+                    "\t", 2
+                )
+                ortho = pynini.accep(ortho_str)
+                hypo_p = pynini.accep(hypo_p_str.replace(" ", ""))
+                gold_p = pynini.accep(gold_p_str.replace(" ", ""))
+                if rewrite.matches(ortho, hypo_p, cg_fst):
+                    if gold_p == hypo_p:
+                        rulematch_predmatch += 1
+                    else:
+                        rulematch_pred_notmatch += 1
                 else:
-                    not_rulematch_pred_notmatch += 1
-            elif act == pred_pron:
-                rulematch_predmatch += 1
-            else:
-                rulematch_pred_notmatch += 1
-    # Collects percentages.
-    rule_m_pred_nm = 100 * rulematch_pred_notmatch / total_records
-    rule_m_pred_m = 100 * rulematch_predmatch / total_records
-    rule_nm_pred_m = 100 * not_rulematch_predmatch / total_records
-    rule_nm_pred_nm = 100 * not_rulematch_pred_notmatch / total_records
-    # Builds and prints the table.
-    print_table = prettytable.PrettyTable()
-    print_table.field_names = ["", "CG match", "CG non-match"]
-    print_table.add_row(
-        ["Pron match", f"{rule_m_pred_m:.2f}", f"{rule_nm_pred_m:.2f}"]
-    )
-    print_table.add_row(
-        ["Pron non-match", f"{rule_m_pred_nm:.2f}", f"{rule_nm_pred_nm:.2f}"]
-    )
-    print(print_table)
+                    if gold_p == hypo_p:
+                        not_rulematch_predmatch += 1
+                    else:
+                        not_rulematch_pred_notmatch += 1
+
```

Comment:
Please remove blank line in function. It makes it hard to tell where the function starts and ends IMO.

Reply:
Fixed.